extensions/machine-learning-services/src/common/queryRunner.ts

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as azdata from 'azdata';
+import * as nbExtensionApis from '../typings/notebookServices';
+import { ApiWrapper } from './apiWrapper';
+import { log } from 'util';
+
+const listPythonPackagesQuery = `
+EXEC sp_execute_external_script
+@language=N'Python',
+@script=N'import pkg_resources
+import pandas
+OutputDataSet = pandas.DataFrame([(d.project_name, d.version) for d in pkg_resources.working_set])'
+`;
+
+const checkMlInstalledQuery = `
+Declare @tablevar table(name NVARCHAR(MAX), min INT, max INT, config_value bit, run_value bit)
+insert into @tablevar(name, min, max, config_value, run_value) exec sp_configure
+
+Declare @external_script_enabled bit
+SELECT @external_script_enabled=config_value FROM @tablevar WHERE name = 'external scripts enabled'
+SELECT @external_script_enabled`;
+
+const checkPythonInstalledQuery = `
+
+SELECT is_installed
+FROM sys.dm_db_external_language_stats s, sys.external_languages l
+WHERE s.external_language_id = l.external_language_id AND language = 'Python'`;
+
+const modifyExternalScriptConfigQuery = `
+
+EXEC sp_configure 'external scripts enabled', #CONFIG_VALUE#;
+RECONFIGURE WITH OVERRIDE;
+
+Declare @tablevar table(name NVARCHAR(MAX), min INT, max INT, config_value bit, run_value bit)
+insert into @tablevar(name, min, max, config_value, run_value) exec sp_configure
+
+Declare @external_script_enabled bit
+SELECT @external_script_enabled=config_value FROM @tablevar WHERE name = 'external scripts enabled'
+SELECT @external_script_enabled`;
+
+/**
+ * SQL Query runner
+ */
+export class QueryRunner {
+
+	constructor(private _apiWrapper: ApiWrapper) {
+	}
+
+	/**
+	 * Returns python packages installed in SQL server instance
+	 * @param connection SQL Connection
+	 */
+	public async getPythonPackages(connection: azdata.connection.ConnectionProfile): Promise<nbExtensionApis.IPackageDetails[]> {
+		let packages = [];
+		let result = await this.runQuery(connection, listPythonPackagesQuery);
+		if (result) {
+			if (result.rows.length > 0) {
```

Comment:
Maybe combine this if case with the one above? Also, do we need to check if rows is undefined?